### PR TITLE
metrics: add field to store metrics in ToolchainStatus

### PR DESCRIPTION
**Pull request by xcoulon**
E2E Tests for https://github.com/codeready-toolchain/host-operator/pull/401
Checks the new `sandbox_users_per_activations ` metric when users are deactivated and reactivated multiple times

### CODE REVIEW

File: test/e2e/nstemplatetier_test.go [hunk 1]
```diff
@@ -100,9 +100,9 @@ func TestUpdateNSTemplateTier(t *testing.T) {
 	// finally, all user namespaces are verified.
 	// So, in this test, we verify that namespace resources and cluster resources are updated, on 2 groups of users with different tiers ;)
 
-	count := 2*MaxPoolSize + 1
+	count := 2*1 + 1
```
Comment (MatousJobanek) [inline test/e2e/nstemplatetier_test.go:103]: is this a valid change?

File: test/e2e/nstemplatetier_test.go [hunk 2]
```diff
@@ -100,9 +100,9 @@ func TestUpdateNSTemplateTier(t *testing.T) {
...
 	ctx, hostAwait, memberAwait, _ := WaitForDeployments(t, &toolchainv1alpha1.NSTemplateTier{})
-	defer ctx.Cleanup()
+	// defer ctx.Cleanup()
```
Comment (MatousJobanek) [inline test/e2e/nstemplatetier_test.go:105]: zombie?
```suggestion
	defer ctx.Cleanup()
```

File: test/e2e/user_management_test.go [hunk 3]
```diff
@@ -38,7 +41,7 @@ func (s *userManagementTestSuite) SetupSuite() {
 }
 
 func (s *userManagementTestSuite) TearDownTest() {
-	s.ctx.Cleanup()
+	// s.ctx.Cleanup()
```
Comment (MatousJobanek) [inline test/e2e/user_management_test.go:44]: zombie?
```suggestion
	s.ctx.Cleanup()
```

File: test/e2e/nstemplatetier_test.go (hunk 1 above)
Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go:103]: that was not intended to be here, I'll revert that one

File: test/e2e/nstemplatetier_test.go (hunk 2 above)
Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go:105]: the usual mistake 🤦‍♂️

File: test/e2e/user_management_test.go (hunk 3 above)
Comment (xcoulon) [reply test/e2e/user_management_test.go]: yes :(

File: test/e2e/nstemplatetier_test.go (hunk 1 above)
Comment (xcoulon) [reply test/e2e/nstemplatetier_test.go:103]: reverted in f1497e6

**Review by MatousJobanek — APPROVED** [review]
Cool :+1:

File: test/e2e/user_management_test.go [hunk 4]
```diff
@@ -205,6 +208,88 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
...
+		metricsAssertion.WaitForMetricDelta(UsersPerActivationMetric, 0, "activations", "2") // (unchanged after other usersignup was deleted)
+		metricsAssertion.WaitForMetricDelta(UsersPerActivationMetric, 0, "activations", "3") // (unchanged after other usersignup was deleted)
+		metricsAssertion.WaitForMetricDelta(UsersPerActivationMetric, 0, "activations", "4") // (unchanged after other usersignup was deleted)
+
+		// when deleting user "user-0002"
+		err = s.hostAwait.Client.Delete(context.TODO(), usersignups["user-0002"])
+
+		// then
+		require.NoError(t, err)
+		// and verify that the values of the `sandbox_users_per_activations` metric were updated accordingly (ie, decremented)
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply test/e2e/user_management_test.go]: it's a left-over from the time where I decremented the counter when a user was deactivated, which is not the case anymore. 
I fixed the comments in a41718f

**Pull request merged**